### PR TITLE
Added missing integration tests for the pow and power functions

### PR DESCRIPTION
**Pull request by matthewryanwells**
### Description
Add integration tests for the pow and power functions which were previously missing it
 
### Issues Resolved
https://github.com/opensearch-project/sql/issues/1190
 
### Check List
- [x] All tests pass, including unit test, integration test and doctest
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by GumpacG** [comment]
Can you also add a test where the result would be `null`

**Comment by matthewryanwells** [comment]
> Can you also add a test where the result would be `null`

Is there a mathematical equation that would cause pow/power to output a null or do you want me to force a null value by giving it invalid data?

**Comment by GumpacG** [comment]
> > Can you also add a test where the result would be `null`
> 
> Is there a mathematical equation that would cause pow/power to output a null or do you want me to force a null value by giving it invalid data?

Some examples here https://stackoverflow.com/questions/20250453/why-math-powa-b-is-nan-when-a-is-a-negative-number-and-b-is-a-non-integer

**Comment by matthewryanwells** [comment]
> > > Can you also add a test where the result would be `null`
> > 
> > 
> > Is there a mathematical equation that would cause pow/power to output a null or do you want me to force a null value by giving it invalid data?
> 
> Some examples here https://stackoverflow.com/questions/20250453/why-math-powa-b-is-nan-when-a-is-a-negative-number-and-b-is-a-non-integer

From what I have seen with the other tests IT don't actually ever test for null or invalid values, those are testing in the unit tests. Also if you try to input data that would cause an invalid outcome the tests returns an `IllegalArgumentException`

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by GumpacG — APPROVED** [review]
LGTM! Although, some GH actions are failing.

File: core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java [hunk 1]
```diff
@@ -1570,6 +1570,50 @@ public void pow_null_missing() {
...
+    assertEquals(pow.type(), DOUBLE);
+    assertEquals(String.format("pow(%s, %s)", (double) -2, 1.5), pow.toString());
+    assertTrue(pow.valueOf(valueEnv()).isNull());
+
+    pow = DSL.pow(DSL.literal((double) -2), DSL.literal((double) 2));
+    assertEquals(pow.type(), DOUBLE);
+    assertEquals(String.format("pow(%s, %s)",(double) -2, (double) 2), pow.toString());
+    assertThat(
+            pow.valueOf(valueEnv()),
+            allOf(hasType(DOUBLE), hasValue(Math.pow(-2, 2))));
```
Comment (GumpacG) [inline core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java:1588]: I think cases that have non-null results should be in a different test or removed if they already exist

Comment (matthewryanwells) [reply core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java:1599]: Separated tests in `16661a0`
